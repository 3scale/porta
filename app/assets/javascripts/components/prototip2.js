

var Bubble = Class.create();
Bubble.prototype = {
  initialize: function(el) {
    this.offsetX = -8;
    this.offsetY = 38;

    this.target = el;
    // this.text = text;
    this.makeBubble();
  },
 
  makeBubble: function() {
    var bubble = $("bubble_" + this.target.title);
    // var bubble = new Element('div', { 'class': 'bubble'});
    // var content = new Element('div', { 'class': 'content'});
    // content.innerHTML = this.text;

    // bubble.appendChild(content);
    this.bubble = bubble;
    // new Insertion.Bottom(body, bubble);
    // bubble.hide();
    
    
    
    // set position

    // listen!
    
    this.target.observe('mouseover', (
        function() { 

          var position = this.target.cumulativeOffset();

          this.bubble.style.top = position[1] + this.offsetX + "px";
          this.bubble.style.left = position[0] + this.target.getWidth() + this.offsetY + "px";
          
          this.bubble.show(); 
        }).bind(this));
        
    this.target.observe('mouseout', (
        function() { 
          this.bubble.hide(); 
        }).bind(this));
        
  }
  
  
}

var BubbleHelper = {
  set: function(){
    $$('.define').each(function(l){
      new Bubble(l);      
    });
}}
  //     function(event) { 
  // 
  //       if ((event.element().className == 'submit' )) {
  // 
  //         var form = event.element().up('form');
  //         var formData = form.serialize(true);
  //         var url = form.action;
  //         
  //         this.ajaxPost(formData, url);
  //         event.stop();
  //       }
  //     })
  //   )        
  //   
  // }
