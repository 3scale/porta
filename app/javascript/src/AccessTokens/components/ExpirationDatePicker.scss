@import '~@patternfly/patternfly/patternfly-addons.css';

.pf-c-calendar-month,
.pf-c-form-control-expiration {
  width: 50%;
}

.pf-c-form-control-expiration {
  margin-right: 1em;
}

button.pf-c-form__group-label-help {
  min-width: auto;
  width: auto;
}
