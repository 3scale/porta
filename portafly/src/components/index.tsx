export * from './pages'
export * from './shared'
export * from './AppLayout'
export * from './AppNavExpandable'
export * from './AppNavGroup'
export * from './AppNavItem'
export * from './AppHeaderTools'
export * from './LazyRoute'
export * from './NotFound'
export * from './SwitchWith404'
export * from './util'
export * from './Root'
