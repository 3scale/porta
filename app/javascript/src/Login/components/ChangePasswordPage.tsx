import {
  ActionGroup,
  Button,
  Form,
  FormGroup,
  LoginPage,
  TextInput
} from '@patternfly/react-core'
import { useState } from 'react'

import { createReactWrapper } from 'utilities/createReactWrapper'
import brandImg from 'Login/assets/images/3scale_Logo_Reverse.png'
import PF4DownstreamBG from 'Login/assets/images/PF4DownstreamBG.svg'
import { CSRFToken } from 'utilities/CSRFToken'
import { validateChangePassword } from 'Login/utils/validations'
import { LoginAlert } from 'Login/components/LoginAlert'

import type { FunctionComponent } from 'react'
import type { FlashMessage } from 'Types'

interface Props {
  lostPasswordToken?: string | null;
  url?: string;
  flashMessages: FlashMessage[];
}

const ChangePasswordPage: FunctionComponent<Props> = ({
  lostPasswordToken = null,
  url,
  flashMessages
}) => {
  const [state, setState] = useState({
    password: '',
    passwordConfirmation: ''
  })
  const [validationVisibility, setValidationVisibility] = useState({
    password: false,
    passwordConfirmation: false
  })

  const handleOnChange = (field: keyof typeof state) => {
    return (value: string) => {
      setState(prev => ({ ...prev, [field]: value }))
      setValidationVisibility(prev => ({ ...prev, [field]: false }))
    }
  }

  const handleOnBlur = (field: keyof typeof state) => {
    return () => {
      setValidationVisibility(prev => ({ ...prev, [field]: true }))
    }
  }

  const alert = flashMessages.length ? flashMessages[0] : undefined

  const validation = validateChangePassword(state)

  const passwordErrors = validation?.password
  const passwordConfirmationErrors = validation?.passwordConfirmation

  const passwordValidated = (validationVisibility.password && passwordErrors) ? 'error' : 'default'
  const passwordConfirmationValidated = (validationVisibility.passwordConfirmation && passwordConfirmationErrors) ? 'error' : 'default'

  return (
    <LoginPage
      backgroundImgAlt="Red Hat 3scale API Management"
      backgroundImgSrc={PF4DownstreamBG}
      brandImgAlt="Red Hat 3scale API Management"
      brandImgSrc={brandImg}
      loginTitle="Change Password"
    >
      <Form
        noValidate
        acceptCharset="UTF-8"
        action={url}
        id="edit_user_2"
        method="post"
      >
        <LoginAlert message={alert?.message} type={alert?.type} />

        <input name="_method" type="hidden" value="put" />
        <input name="utf8" type="hidden" value="âœ“" />
        <CSRFToken />

        <FormGroup
          isRequired
          autoComplete="off"
          fieldId="user_password"
          helperTextInvalid={passwordErrors?.[0]}
          label="Password"
          validated={passwordValidated}
        >
          <TextInput
            autoFocus
            isRequired
            autoComplete="off"
            id="user_password"
            name="user[password]"
            type="password"
            validated={passwordValidated}
            value={state.password}
            onBlur={handleOnBlur('password')}
            onChange={handleOnChange('password')}
          />
        </FormGroup>

        <FormGroup
          isRequired
          autoComplete="off"
          fieldId="user_password_confirmation"
          helperTextInvalid={passwordConfirmationErrors?.[0]}
          label="Password confirmation"
          validated={passwordConfirmationValidated}
        >
          <TextInput
            isRequired
            autoComplete="off"
            id="user_password_confirmation"
            name="user[password_confirmation]"
            type="password"
            validated={passwordConfirmationValidated}
            value={state.passwordConfirmation}
            onBlur={handleOnBlur('passwordConfirmation')}
            onChange={handleOnChange('passwordConfirmation')}
          />
        </FormGroup>

        {!!lostPasswordToken && <input id="password_reset_token" name="password_reset_token" type="hidden" value={lostPasswordToken} />}
        <ActionGroup>
          <Button
            isBlock
            isDisabled={validation !== undefined}
            type="submit"
            variant="primary"
          >
            Change Password
          </Button>
        </ActionGroup>
      </Form>
    </LoginPage>
  )
}

// eslint-disable-next-line react/jsx-props-no-spreading
const ChangePasswordWrapper = (props: Props, containerId: string): void => { createReactWrapper(<ChangePasswordPage {...props} />, containerId) }

export type { Props }
export { ChangePasswordPage, ChangePasswordWrapper }
