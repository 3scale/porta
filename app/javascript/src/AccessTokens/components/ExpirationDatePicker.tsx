import { useState, useMemo } from 'react'
import { Alert, CalendarMonth, FormGroup, FormSelect, FormSelectOption } from '@patternfly/react-core'

import { createReactWrapper } from 'utilities/createReactWrapper'

import type { FunctionComponent, FormEvent } from 'react'

import './ExpirationDatePicker.scss'

interface ExpirationItem {
  id: string;
  label: string;
  period: number; // In seconds
}

const collection: ExpirationItem[] = [
  { id: '7', label: '7 days', period: 7 },
  { id: '30', label: '30 days', period: 30 },
  { id: '60', label: '60 days', period: 60 },
  { id: '90', label: '90 days', period: 90 },
  { id: 'custom', label: 'Custom...', period: 0 },
  { id: 'no-exp', label: 'No expiration', period: 0 }
]

const dayMs = 60 * 60 * 24 * 1000

interface Props {
  id: string;
  label: string | null;
}

const ExpirationDatePicker: FunctionComponent<Props> = ({ id, label }) => {
  const [selectedItem, setSelectedItem] = useState(collection[0])
  const [pickedDate, setPickedDate] = useState(new Date())
  const fieldName = `human_${id}`
  const fieldLabel = label ?? 'Expires in'

  const fieldDate = useMemo(() => {
    if (selectedItem.period === 0) return null

    return new Date(new Date().getTime() + selectedItem.period * dayMs)
  }, [selectedItem])

  const fieldHint = useMemo(() => {
    if (!fieldDate) return

    const date = new Date(fieldDate)
    date.setHours(0, 0, 0, 0)

    return `The token will expire on ${date.toLocaleDateString()}`
  }, [fieldDate])

  const dateValue = useMemo(() => {
    let value = ''

    if (fieldDate) {
      value = fieldDate.toISOString()
    } else if (selectedItem.id === 'custom' ) {
      value = pickedDate.toISOString()
    }

    return value
  }, [fieldDate, selectedItem, pickedDate])

  const handleOnChange = (_value: string, event: FormEvent<HTMLSelectElement>) => {
    const value = (event.target as HTMLSelectElement).value
    const selected = collection.find(i => i.id === value) ?? null

    if (selected === null) return

    setSelectedItem(selected)
    setPickedDate(new Date())
  }

  return (
    <>
      <FormGroup
        isRequired
        fieldId={fieldName}
        label={fieldLabel}
      >
        <FormSelect
          className="pf-c-form-control-expiration"
          id={fieldName}
          value={selectedItem.id}
          onChange={handleOnChange}
        >
          {collection.map((item: ExpirationItem) => {
            return (
              <FormSelectOption
                key={item.id}
                label={item.label}
                value={item.id}
              />
            )
          })}
        </FormSelect>
        <span className="pf-c-form-control-expiration-hint">{fieldHint}</span>
      </FormGroup>
      <input id={id} name={id} type="hidden" value={dateValue} />
      {selectedItem.id === 'custom' && (
        <CalendarMonth className="pf-u-mt-md" date={pickedDate} onChange={setPickedDate} />
      )}
      {dateValue === '' && (
        <>
          <br />
          <Alert title="Expiration is recommended" variant="warning">
            It is strongly recommended that you set an expiration date for your token to help keep your information
            secure
          </Alert>
        </>
      )}
    </>
  )
}

const ExpirationDatePickerWrapper = (props: Props, containerId: string): void => { createReactWrapper(<ExpirationDatePicker id={props.id} label={props.label} />, containerId) }

export type { ExpirationItem, Props }
export { ExpirationDatePicker, ExpirationDatePickerWrapper }
