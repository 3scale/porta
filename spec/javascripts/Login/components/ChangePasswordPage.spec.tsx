import { mount } from 'enzyme'

import { ChangePasswordPage } from 'Login/components/ChangePasswordPage'

import type { Props } from 'Login/components/ChangePasswordPage'

const defaultProps: Props = {
  lostPasswordToken: 'foo',
  url: 'foo/bar',
  flashMessages: []
}

const mountWrapper = (props: Partial<Props> = {}) => mount(<ChangePasswordPage {...{ ...defaultProps, ...props }} />)

it('should render itself', () => {
  const wrapper = mountWrapper()
  expect(wrapper).toMatchSnapshot()
})

it('should render with server side errors when present', () => {
  const wrapper = mountWrapper({ flashMessages: [{ type: 'error', message: 'Ooops!' }] })
  expect(wrapper).toMatchSnapshot()
})

// TODO: Test Form errors and interaction.
