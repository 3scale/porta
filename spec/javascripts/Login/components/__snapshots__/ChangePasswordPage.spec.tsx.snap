// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render itself 1`] = `
<ChangePasswordPage
  alerts={[]}
  lostPasswordToken="foo"
  url="foo/bar"
>
  <LoginPage
    backgroundImgAlt="Red Hat 3scale API Management"
    backgroundImgSrc=""
    brandImgAlt="Red Hat 3scale API Management"
    brandImgSrc=""
    loginTitle="Change Password"
  >
    <Login
      className=""
      footer={
        <LoginFooter>
          <p>
            
          </p>
          <List />
        </LoginFooter>
      }
      header={
        <LoginHeader
          headerBrand={
            <React.Fragment>
              <Brand
                alt="Red Hat 3scale API Management"
                src=""
              />
            </React.Fragment>
          }
        />
      }
    >
      <div
        className="pf-c-login"
      >
        <div
          className="pf-c-login__container"
        >
          <LoginHeader
            headerBrand={
              <React.Fragment>
                <Brand
                  alt="Red Hat 3scale API Management"
                  src=""
                />
              </React.Fragment>
            }
          >
            <header
              className="pf-c-login__header"
            >
              <Brand
                alt="Red Hat 3scale API Management"
                src=""
              >
                <img
                  alt="Red Hat 3scale API Management"
                  className="pf-c-brand"
                  src=""
                />
              </Brand>
            </header>
          </LoginHeader>
          <main
            className="pf-c-login__main"
          >
            <LoginMainHeader
              title="Change Password"
            >
              <header
                className="pf-c-login__main-header"
              >
                <Title
                  headingLevel="h2"
                  size="3xl"
                >
                  <h2
                    className="pf-c-title pf-m-3xl"
                    data-ouia-component-id="OUIA-Generated-Title-1"
                    data-ouia-component-type="PF4/Title"
                    data-ouia-safe={true}
                  >
                    Change Password
                  </h2>
                </Title>
              </header>
            </LoginMainHeader>
            <LoginMainBody>
              <div
                className="pf-c-login__main-body"
              >
                <Form
                  acceptCharset="UTF-8"
                  action="foo/bar"
                  id="edit_user_2"
                  method="post"
                  noValidate={true}
                >
                  <FormBase
                    acceptCharset="UTF-8"
                    action="foo/bar"
                    id="edit_user_2"
                    innerRef={null}
                    method="post"
                    noValidate={true}
                  >
                    <form
                      acceptCharset="UTF-8"
                      action="foo/bar"
                      className="pf-c-form"
                      id="edit_user_2"
                      method="post"
                      noValidate={true}
                    >
                      <LoginAlert>
                        <Alert
                          className="invisible"
                          isInline={true}
                          variant="default"
                        >
                          <div
                            aria-label="Default Alert"
                            className="pf-c-alert pf-m-inline invisible"
                            data-ouia-component-id="OUIA-Generated-Alert-default-1"
                            data-ouia-component-type="PF4/Alert"
                            data-ouia-safe={true}
                            onMouseEnter={[Function]}
                            onMouseLeave={[Function]}
                          >
                            <AlertIcon
                              variant="default"
                            >
                              <div
                                className="pf-c-alert__icon"
                              >
                                <BellIcon
                                  color="currentColor"
                                  noVerticalAlign={false}
                                  size="sm"
                                >
                                  <svg
                                    aria-hidden={true}
                                    aria-labelledby={null}
                                    fill="currentColor"
                                    height="1em"
                                    role="img"
                                    style={
                                      {
                                        "verticalAlign": "-0.125em",
                                      }
                                    }
                                    viewBox="0 0 896 1024"
                                    width="1em"
                                  >
                                    <path
                                      d="M448,0 C465.333333,0 480.333333,6.33333333 493,19 C505.666667,31.6666667 512,46.6666667 512,64 L512,106 L514.23,106.45 C587.89,121.39 648.48,157.24 696,214 C744,271.333333 768,338.666667 768,416 C768,500 780,568.666667 804,622 C818.666667,652.666667 841.333333,684 872,716 C873.773676,718.829136 875.780658,721.505113 878,724 C890,737.333333 896,752.333333 896,769 C896,785.666667 890,800.333333 878,813 C866,825.666667 850.666667,832 832,832 L63.3,832 C44.9533333,831.84 29.8533333,825.506667 18,813 C6,800.333333 0,785.666667 0,769 C0,752.333333 6,737.333333 18,724 L24,716 L25.06,714.9 C55.1933333,683.28 77.5066667,652.313333 92,622 C116,568.666667 128,500 128,416 C128,338.666667 152,271.333333 200,214 C248,156.666667 309.333333,120.666667 384,106 L384,63.31 C384.166667,46.27 390.5,31.5 403,19 C415.666667,6.33333333 430.666667,0 448,0 Z M576,896 L576,897.08 C575.74,932.6 563.073333,962.573333 538,987 C512.666667,1011.66667 482.666667,1024 448,1024 C413.333333,1024 383.333333,1011.66667 358,987 C332.666667,962.333333 320,932 320,896 L576,896 Z"
                                    />
                                  </svg>
                                </BellIcon>
                              </div>
                            </AlertIcon>
                            <h4
                              className="pf-c-alert__title"
                            >
                              <span
                                className="pf-u-screen-reader"
                              >
                                Default alert:
                              </span>
                            </h4>
                          </div>
                        </Alert>
                      </LoginAlert>
                      <input
                        name="_method"
                        type="hidden"
                        value="put"
                      />
                      <input
                        name="utf8"
                        type="hidden"
                        value="âœ“"
                      />
                      <CSRFToken>
                        <input
                          name="mocked-csrf-param"
                          type="hidden"
                          value="mocked-csrf-token"
                        />
                      </CSRFToken>
                      <FormGroup
                        autoComplete="off"
                        fieldId="user_password"
                        helperTextInvalid="Password is mandatory"
                        isRequired={true}
                        label="Password"
                        validated="default"
                      >
                        <GenerateId
                          prefix="pf-random-id-"
                        >
                          <div
                            autoComplete="off"
                            className="pf-c-form__group"
                          >
                            <div
                              className="pf-c-form__group-label"
                            >
                              <label
                                className="pf-c-form__label"
                                htmlFor="user_password"
                              >
                                <span
                                  className="pf-c-form__label-text"
                                >
                                  Password
                                </span>
                                <span
                                  aria-hidden="true"
                                  className="pf-c-form__label-required"
                                >
                                   
                                  *
                                </span>
                              </label>
                               
                            </div>
                            <div
                              className="pf-c-form__group-control"
                            >
                              <TextInput
                                autoComplete="off"
                                autoFocus={true}
                                id="user_password"
                                isRequired={true}
                                name="user[password]"
                                onBlur={[Function]}
                                onChange={[Function]}
                                type="password"
                                validated="default"
                                value=""
                              >
                                <TextInputBase
                                  aria-label={null}
                                  autoComplete="off"
                                  autoFocus={true}
                                  className=""
                                  id="user_password"
                                  innerRef={null}
                                  isDisabled={false}
                                  isIconSprite={false}
                                  isLeftTruncated={false}
                                  isReadOnly={false}
                                  isRequired={true}
                                  name="user[password]"
                                  onBlur={[Function]}
                                  onChange={[Function]}
                                  ouiaSafe={true}
                                  type="password"
                                  validated="default"
                                  value=""
                                >
                                  <input
                                    aria-invalid={false}
                                    aria-label={null}
                                    autoComplete="off"
                                    autoFocus={true}
                                    className="pf-c-form-control"
                                    data-ouia-component-id="OUIA-Generated-TextInputBase-1"
                                    data-ouia-component-type="PF4/TextInput"
                                    data-ouia-safe={true}
                                    disabled={false}
                                    id="user_password"
                                    name="user[password]"
                                    onBlur={[Function]}
                                    onChange={[Function]}
                                    onFocus={[Function]}
                                    required={true}
                                    type="password"
                                    value=""
                                  />
                                </TextInputBase>
                              </TextInput>
                            </div>
                          </div>
                        </GenerateId>
                      </FormGroup>
                      <FormGroup
                        autoComplete="off"
                        fieldId="user_password_confirmation"
                        helperTextInvalid="Password confirmation is mandatory"
                        isRequired={true}
                        label="Password confirmation"
                        validated="default"
                      >
                        <GenerateId
                          prefix="pf-random-id-"
                        >
                          <div
                            autoComplete="off"
                            className="pf-c-form__group"
                          >
                            <div
                              className="pf-c-form__group-label"
                            >
                              <label
                                className="pf-c-form__label"
                                htmlFor="user_password_confirmation"
                              >
                                <span
                                  className="pf-c-form__label-text"
                                >
                                  Password confirmation
                                </span>
                                <span
                                  aria-hidden="true"
                                  className="pf-c-form__label-required"
                                >
                                   
                                  *
                                </span>
                              </label>
                               
                            </div>
                            <div
                              className="pf-c-form__group-control"
                            >
                              <TextInput
                                autoComplete="off"
                                id="user_password_confirmation"
                                isRequired={true}
                                name="user[password_confirmation]"
                                onBlur={[Function]}
                                onChange={[Function]}
                                type="password"
                                validated="default"
                                value=""
                              >
                                <TextInputBase
                                  aria-label={null}
                                  autoComplete="off"
                                  className=""
                                  id="user_password_confirmation"
                                  innerRef={null}
                                  isDisabled={false}
                                  isIconSprite={false}
                                  isLeftTruncated={false}
                                  isReadOnly={false}
                                  isRequired={true}
                                  name="user[password_confirmation]"
                                  onBlur={[Function]}
                                  onChange={[Function]}
                                  ouiaSafe={true}
                                  type="password"
                                  validated="default"
                                  value=""
                                >
                                  <input
                                    aria-invalid={false}
                                    aria-label={null}
                                    autoComplete="off"
                                    className="pf-c-form-control"
                                    data-ouia-component-id="OUIA-Generated-TextInputBase-2"
                                    data-ouia-component-type="PF4/TextInput"
                                    data-ouia-safe={true}
                                    disabled={false}
                                    id="user_password_confirmation"
                                    name="user[password_confirmation]"
                                    onBlur={[Function]}
                                    onChange={[Function]}
                                    onFocus={[Function]}
                                    required={true}
                                    type="password"
                                    value=""
                                  />
                                </TextInputBase>
                              </TextInput>
                            </div>
                          </div>
                        </GenerateId>
                      </FormGroup>
                      <input
                        id="password_reset_token"
                        name="password_reset_token"
                        type="hidden"
                        value="foo"
                      />
                      <ActionGroup>
                        <div
                          className="pf-c-form__group pf-m-action"
                        >
                          <div
                            className="pf-c-form__group-control"
                          >
                            <div
                              className="pf-c-form__actions"
                            >
                              <Button
                                isBlock={true}
                                isDisabled={true}
                                type="submit"
                                variant="primary"
                              >
                                <ButtonBase
                                  innerRef={null}
                                  isBlock={true}
                                  isDisabled={true}
                                  type="submit"
                                  variant="primary"
                                >
                                  <button
                                    aria-disabled={true}
                                    aria-label={null}
                                    className="pf-c-button pf-m-primary pf-m-block pf-m-disabled"
                                    data-ouia-component-id="OUIA-Generated-Button-primary-1"
                                    data-ouia-component-type="PF4/Button"
                                    data-ouia-safe={true}
                                    disabled={true}
                                    role={null}
                                    tabIndex={null}
                                    type="submit"
                                  >
                                    Change Password
                                  </button>
                                </ButtonBase>
                              </Button>
                            </div>
                          </div>
                        </div>
                      </ActionGroup>
                    </form>
                  </FormBase>
                </Form>
              </div>
            </LoginMainBody>
          </main>
          <LoginFooter>
            <footer
              className="pf-c-login__footer"
            >
              <p />
              <List>
                <ul
                  className="pf-c-list"
                />
              </List>
            </footer>
          </LoginFooter>
        </div>
      </div>
    </Login>
  </LoginPage>
</ChangePasswordPage>
`;

exports[`should render with server side errors when present 1`] = `
<ChangePasswordPage
  alerts={
    [
      {
        "message": "Ooops!",
        "type": "danger",
      },
    ]
  }
  lostPasswordToken="foo"
  url="foo/bar"
>
  <LoginPage
    backgroundImgAlt="Red Hat 3scale API Management"
    backgroundImgSrc=""
    brandImgAlt="Red Hat 3scale API Management"
    brandImgSrc=""
    loginTitle="Change Password"
  >
    <Login
      className=""
      footer={
        <LoginFooter>
          <p>
            
          </p>
          <List />
        </LoginFooter>
      }
      header={
        <LoginHeader
          headerBrand={
            <React.Fragment>
              <Brand
                alt="Red Hat 3scale API Management"
                src=""
              />
            </React.Fragment>
          }
        />
      }
    >
      <div
        className="pf-c-login"
      >
        <div
          className="pf-c-login__container"
        >
          <LoginHeader
            headerBrand={
              <React.Fragment>
                <Brand
                  alt="Red Hat 3scale API Management"
                  src=""
                />
              </React.Fragment>
            }
          >
            <header
              className="pf-c-login__header"
            >
              <Brand
                alt="Red Hat 3scale API Management"
                src=""
              >
                <img
                  alt="Red Hat 3scale API Management"
                  className="pf-c-brand"
                  src=""
                />
              </Brand>
            </header>
          </LoginHeader>
          <main
            className="pf-c-login__main"
          >
            <LoginMainHeader
              title="Change Password"
            >
              <header
                className="pf-c-login__main-header"
              >
                <Title
                  headingLevel="h2"
                  size="3xl"
                >
                  <h2
                    className="pf-c-title pf-m-3xl"
                    data-ouia-component-id="OUIA-Generated-Title-2"
                    data-ouia-component-type="PF4/Title"
                    data-ouia-safe={true}
                  >
                    Change Password
                  </h2>
                </Title>
              </header>
            </LoginMainHeader>
            <LoginMainBody>
              <div
                className="pf-c-login__main-body"
              >
                <Form
                  acceptCharset="UTF-8"
                  action="foo/bar"
                  id="edit_user_2"
                  method="post"
                  noValidate={true}
                >
                  <FormBase
                    acceptCharset="UTF-8"
                    action="foo/bar"
                    id="edit_user_2"
                    innerRef={null}
                    method="post"
                    noValidate={true}
                  >
                    <form
                      acceptCharset="UTF-8"
                      action="foo/bar"
                      className="pf-c-form"
                      id="edit_user_2"
                      method="post"
                      noValidate={true}
                    >
                      <LoginAlert
                        message="Ooops!"
                        type="danger"
                      >
                        <Alert
                          className=""
                          isInline={true}
                          title={
                            [
                              <p>
                                Ooops!
                              </p>,
                            ]
                          }
                          variant="danger"
                        >
                          <div
                            aria-label="Danger Alert"
                            className="pf-c-alert pf-m-inline pf-m-danger"
                            data-ouia-component-id="OUIA-Generated-Alert-danger-1"
                            data-ouia-component-type="PF4/Alert"
                            data-ouia-safe={true}
                            onMouseEnter={[Function]}
                            onMouseLeave={[Function]}
                          >
                            <AlertIcon
                              variant="danger"
                            >
                              <div
                                className="pf-c-alert__icon"
                              >
                                <ExclamationCircleIcon
                                  color="currentColor"
                                  noVerticalAlign={false}
                                  size="sm"
                                >
                                  <svg
                                    aria-hidden={true}
                                    aria-labelledby={null}
                                    fill="currentColor"
                                    height="1em"
                                    role="img"
                                    style={
                                      {
                                        "verticalAlign": "-0.125em",
                                      }
                                    }
                                    viewBox="0 0 512 512"
                                    width="1em"
                                  >
                                    <path
                                      d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                                    />
                                  </svg>
                                </ExclamationCircleIcon>
                              </div>
                            </AlertIcon>
                            <h4
                              className="pf-c-alert__title"
                            >
                              <span
                                className="pf-u-screen-reader"
                              >
                                Danger alert:
                              </span>
                              <p
                                key="0"
                              >
                                Ooops!
                              </p>
                            </h4>
                          </div>
                        </Alert>
                      </LoginAlert>
                      <input
                        name="_method"
                        type="hidden"
                        value="put"
                      />
                      <input
                        name="utf8"
                        type="hidden"
                        value="âœ“"
                      />
                      <CSRFToken>
                        <input
                          name="mocked-csrf-param"
                          type="hidden"
                          value="mocked-csrf-token"
                        />
                      </CSRFToken>
                      <FormGroup
                        autoComplete="off"
                        fieldId="user_password"
                        helperTextInvalid="Password is mandatory"
                        isRequired={true}
                        label="Password"
                        validated="default"
                      >
                        <GenerateId
                          prefix="pf-random-id-"
                        >
                          <div
                            autoComplete="off"
                            className="pf-c-form__group"
                          >
                            <div
                              className="pf-c-form__group-label"
                            >
                              <label
                                className="pf-c-form__label"
                                htmlFor="user_password"
                              >
                                <span
                                  className="pf-c-form__label-text"
                                >
                                  Password
                                </span>
                                <span
                                  aria-hidden="true"
                                  className="pf-c-form__label-required"
                                >
                                   
                                  *
                                </span>
                              </label>
                               
                            </div>
                            <div
                              className="pf-c-form__group-control"
                            >
                              <TextInput
                                autoComplete="off"
                                autoFocus={true}
                                id="user_password"
                                isRequired={true}
                                name="user[password]"
                                onBlur={[Function]}
                                onChange={[Function]}
                                type="password"
                                validated="default"
                                value=""
                              >
                                <TextInputBase
                                  aria-label={null}
                                  autoComplete="off"
                                  autoFocus={true}
                                  className=""
                                  id="user_password"
                                  innerRef={null}
                                  isDisabled={false}
                                  isIconSprite={false}
                                  isLeftTruncated={false}
                                  isReadOnly={false}
                                  isRequired={true}
                                  name="user[password]"
                                  onBlur={[Function]}
                                  onChange={[Function]}
                                  ouiaSafe={true}
                                  type="password"
                                  validated="default"
                                  value=""
                                >
                                  <input
                                    aria-invalid={false}
                                    aria-label={null}
                                    autoComplete="off"
                                    autoFocus={true}
                                    className="pf-c-form-control"
                                    data-ouia-component-id="OUIA-Generated-TextInputBase-3"
                                    data-ouia-component-type="PF4/TextInput"
                                    data-ouia-safe={true}
                                    disabled={false}
                                    id="user_password"
                                    name="user[password]"
                                    onBlur={[Function]}
                                    onChange={[Function]}
                                    onFocus={[Function]}
                                    required={true}
                                    type="password"
                                    value=""
                                  />
                                </TextInputBase>
                              </TextInput>
                            </div>
                          </div>
                        </GenerateId>
                      </FormGroup>
                      <FormGroup
                        autoComplete="off"
                        fieldId="user_password_confirmation"
                        helperTextInvalid="Password confirmation is mandatory"
                        isRequired={true}
                        label="Password confirmation"
                        validated="default"
                      >
                        <GenerateId
                          prefix="pf-random-id-"
                        >
                          <div
                            autoComplete="off"
                            className="pf-c-form__group"
                          >
                            <div
                              className="pf-c-form__group-label"
                            >
                              <label
                                className="pf-c-form__label"
                                htmlFor="user_password_confirmation"
                              >
                                <span
                                  className="pf-c-form__label-text"
                                >
                                  Password confirmation
                                </span>
                                <span
                                  aria-hidden="true"
                                  className="pf-c-form__label-required"
                                >
                                   
                                  *
                                </span>
                              </label>
                               
                            </div>
                            <div
                              className="pf-c-form__group-control"
                            >
                              <TextInput
                                autoComplete="off"
                                id="user_password_confirmation"
                                isRequired={true}
                                name="user[password_confirmation]"
                                onBlur={[Function]}
                                onChange={[Function]}
                                type="password"
                                validated="default"
                                value=""
                              >
                                <TextInputBase
                                  aria-label={null}
                                  autoComplete="off"
                                  className=""
                                  id="user_password_confirmation"
                                  innerRef={null}
                                  isDisabled={false}
                                  isIconSprite={false}
                                  isLeftTruncated={false}
                                  isReadOnly={false}
                                  isRequired={true}
                                  name="user[password_confirmation]"
                                  onBlur={[Function]}
                                  onChange={[Function]}
                                  ouiaSafe={true}
                                  type="password"
                                  validated="default"
                                  value=""
                                >
                                  <input
                                    aria-invalid={false}
                                    aria-label={null}
                                    autoComplete="off"
                                    className="pf-c-form-control"
                                    data-ouia-component-id="OUIA-Generated-TextInputBase-4"
                                    data-ouia-component-type="PF4/TextInput"
                                    data-ouia-safe={true}
                                    disabled={false}
                                    id="user_password_confirmation"
                                    name="user[password_confirmation]"
                                    onBlur={[Function]}
                                    onChange={[Function]}
                                    onFocus={[Function]}
                                    required={true}
                                    type="password"
                                    value=""
                                  />
                                </TextInputBase>
                              </TextInput>
                            </div>
                          </div>
                        </GenerateId>
                      </FormGroup>
                      <input
                        id="password_reset_token"
                        name="password_reset_token"
                        type="hidden"
                        value="foo"
                      />
                      <ActionGroup>
                        <div
                          className="pf-c-form__group pf-m-action"
                        >
                          <div
                            className="pf-c-form__group-control"
                          >
                            <div
                              className="pf-c-form__actions"
                            >
                              <Button
                                isBlock={true}
                                isDisabled={true}
                                type="submit"
                                variant="primary"
                              >
                                <ButtonBase
                                  innerRef={null}
                                  isBlock={true}
                                  isDisabled={true}
                                  type="submit"
                                  variant="primary"
                                >
                                  <button
                                    aria-disabled={true}
                                    aria-label={null}
                                    className="pf-c-button pf-m-primary pf-m-block pf-m-disabled"
                                    data-ouia-component-id="OUIA-Generated-Button-primary-2"
                                    data-ouia-component-type="PF4/Button"
                                    data-ouia-safe={true}
                                    disabled={true}
                                    role={null}
                                    tabIndex={null}
                                    type="submit"
                                  >
                                    Change Password
                                  </button>
                                </ButtonBase>
                              </Button>
                            </div>
                          </div>
                        </div>
                      </ActionGroup>
                    </form>
                  </FormBase>
                </Form>
              </div>
            </LoginMainBody>
          </main>
          <LoginFooter>
            <footer
              className="pf-c-login__footer"
            >
              <p />
              <List>
                <ul
                  className="pf-c-list"
                />
              </List>
            </footer>
          </LoginFooter>
        </div>
      </div>
    </Login>
  </LoginPage>
</ChangePasswordPage>
`;
